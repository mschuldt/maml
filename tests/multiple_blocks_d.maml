arduino = Arduino(desktop=True)

#each block will will be executed in the blockchain many times before 
#the next one is sent so each of them delays while the next block
#is sent to the vm

@block(once)
def b1():
    print_i(11)
#    delay(100)
@block(once)    
def b2():
    print_i(22)
#    delay(100)
@block(once)    
def b3():
    print_i(33)
#    delay(100)
@block(once)    
def b4():
    print_i(44)
    die(0)


arduino.send(b1)
arduino.send(b2)
arduino.send(b3)
arduino.send(b4)
