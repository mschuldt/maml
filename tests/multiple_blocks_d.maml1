arduino = Arduino(desktop=True)

#each block will will be executed in the blockchain many times before 
#the next one is sent so each of them delays while the next block
#is sent to the vm

@block
def b1():
    print_i(11)
    delay(1)
@block    
def b2():
    print_i(22)
    delay(1)
@block    
def b3():
    print_i(33)
    delay(1)
@block    
def b4():
    print_i(44)
    die(0)


arduino.send(b1)
arduino.send(b2)
arduino.send(b3)
arduino.send(b4)
