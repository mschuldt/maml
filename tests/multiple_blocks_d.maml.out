Initializing avm...
Ready.

11
22
33
44
bye.
